Benqi-Agents/Large-Pool-Withdrawal/src/agent.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Finding,
+  FindingType,
+  FindingSeverity,
+  TransactionEvent,
+  HandleTransaction,
+  getEthersProvider,
+} from "forta-agent";
+
+import { ethers } from "ethers";
+
+import { COMPTROLLER_IFACE, QITOKEN_IFACE } from "./abi";
+
+import { createFinding } from "./finding";
+
+const COMPTROLLER_ADDR = "0x486Af39519B4Dc9a7fCcd318217352830E8AD9b4";
+
+const THRESHOLD_PERCENTAGE = 0.25;
+
+// Array to track the QiToken pools
+let QITOKENS: Set<string> = new Set<string>();
+
+export const getTotalSupply = async (qiToken: string, blockNumber: any) => {
+  // Create an ethers contract instance to query on-chain data
+  const qiTokenContract = new ethers.Contract(
+    qiToken,
+    QITOKEN_IFACE.format(ethers.utils.FormatTypes.full),
+    getEthersProvider()
+  );
+
+  return await qiTokenContract.totalSupply({ blockTag: blockNumber });
+}
```

Comment:
It seems we don't need a provider call since the QI total supply is constant, see [here](https://snowtrace.io/address/0x8729438eb15e2c8b576fcc6aecda6a148776c0f5#code).
I was also querying it on-chain before I took a look at the contract and noticed that.

Reply:
https://github.com/NethermindEth/Forta-Agents/pull/180#discussion_r833120248

Reply:
You're right!